src/components/calcite-ui-icon/calcite-ui-icon.tsx

```diff
@@ -0,0 +1,172 @@
+import { Component, h, Element, Prop, State, Watch, Host } from "@stencil/core";
+import { CSS } from "./resources";
+import { getElementDir } from "../../utils/dom";
+import { fetchIcon, scaleToPx } from "./utils";
+import { Scale } from "../../interfaces/UIIcon";
+import { Theme } from "../../interfaces/common";
+
+@Component({
+  assetsDirs: ["assets"],
+  tag: "calcite-ui-icon",
+  styleUrl: "calcite-ui-icon.scss",
+  shadow: true
+})
+export class CalciteUIIcon {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+
+  @Element()
+  el: HTMLElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * When true, the icon will be filled.
+   */
+  @Prop({
+    reflect: true,
+    mutable: true
```

Comment:
Does this need to be `mutable`? Seems like it is just set from outside and not inside. Applies to other properties too.

Reply:
yeah we are not styling off the prop either AFAICT so the `reflect` could also probably be left off. You'd probably be fine with the default `@Prop() filled: boolean = false;`